### DISCUSSION TITLE
How I enable Rosetta for new Podman V5? I have M2 mac and I still get an error on starting MSSQL image

### DISCUSSION THREAD

**Discussion by Darioconte87** [containers/podman#22445] [General]
Ciao Team,
I downloaded and install podman v5.0.2 on my Mac M2 and I was tried to run server:2022-latest but without success.

I download the image:
podman pull mcr.microsoft.com/mssql/server:2022-latest

I run the container before:

podman run -d -e "ACCEPT_EULA=Y" -e "MSSQL_SA_PASSWORD=XXXXXX" -p 1433:1433 --name XXX --hostname XXX mcr.microsoft.com/mssql/server:2022-latest

Error:
/opt/mssql/bin/sqlservr: Invalid mapping of address 0x2aaaad74a000 in reserved address space below 0x400000000000. Possible causes:

After That I tried also to put platform parameter:

podman run -d -e "ACCEPT_EULA=Y" -e "MSSQL_SA_PASSWORD=XXXXXX" -p 1433:1433 --name XXX --hostname XXX mcr.microsoft.com/mssql/server:2022-latest --platform linux/amd64

Error:
/opt/mssql/bin/permissions_check.sh: line 60: exec: --: invalid option

I think that rosetta is not working or enable. I don't see this option but I read was enabled of default but doesn't work 

Could you help me to fix this error in order to run that container?

Here my podman config json:
[jsonformatter.txt](https://github.com/containers/podman/files/15043857/jsonformatter.txt)
 
Thanks

**Comment by tnk4on** [comment]
I opened the PR https://github.com/containers/podman/pull/21670, but not yet merged.
You can use qemu-user-static by default, but unfortunately it seems to have failed to start the mssql server.
I run it under my Rosetta environment and it succeeded.

```
% podman machine inspect --format {{.Rosetta}}
true
% podman run -d -e "ACCEPT_EULA=Y" -e "MSSQL_SA_PASSWORD=XXXXXXXXXX" -p 1433:1433 --name mssql mcr.microsoft.com/mssql/server:2022-latest
WARNING: image platform (linux/amd64) does not match the expected platform (linux/arm64)
28e83c747cda9e43158c3096ccadcd47a2d424857e0b872ece68ed2b12fdfb29
% podman logs --tail 10 mssql 
2024-04-20 03:34:49.39 spid29s     Server is listening on [ 'any' <ipv4> 1433] accept sockets 1.
2024-04-20 03:34:49.39 Server      Server is listening on [ ::1 <ipv6> 1434] accept sockets 1.
2024-04-20 03:34:49.39 Server      Server is listening on [ 127.0.0.1 <ipv4> 1434] accept sockets 1.
2024-04-20 03:34:49.39 Server      Dedicated admin connection support was established for listening locally on port 1434.
2024-04-20 03:34:49.39 spid29s     Server is listening on [ ::1 <ipv6> 1431] accept sockets 1.
2024-04-20 03:34:49.39 spid29s     Server is listening on [ 127.0.0.1 <ipv4> 1431] accept sockets 1.
2024-04-20 03:34:49.40 spid29s     Recovery is complete. This is an informational message only. No user action is required.
2024-04-20 03:34:49.40 spid29s     SQL Server is now ready for client connections. This is an informational message; no user action is required.
2024-04-20 03:34:49.61 spid27s     The default language (LCID 0) has been set for engine and full-text services.
2024-04-20 03:34:49.67 spid27s     The tempdb database has 4 data file(s).
% podman ps -a
CONTAINER ID  IMAGE                                       COMMAND               CREATED        STATUS        PORTS                             NAMES
28e83c747cda  mcr.microsoft.com/mssql/server:2022-latest  /opt/mssql/bin/sq...  2 minutes ago  Up 2 minutes  0.0.0.0:1433->1433/tcp, 1433/tcp  mssql
% podman top mssql 
USER        PID         PPID        %CPU        ELAPSED          TTY         TIME        COMMAND
mssql       1           0           0.000       5m35.837365411s  ?           0s          /mnt/rosetta /opt/mssql/bin/sqlservr
mssql       4           1           5.077       5m34.837504797s  ?           17s         /mnt/rosetta /opt/mssql/bin/sqlservr
```

**Reply by Darioconte87** [reply]
Ciao @tnk4on,
now is all clear for me.

I read the thread of PR and I appreciate too much the work is doing to take this feature working on Mac Apple Silicon.
I will wait that your PR will be merged in main release.
Is there already a planned target release when this feature will be deployed and available?

**Reply by tnk4on** [reply]
The PR is labeled v5.1. I am not sure about the detailed release plan .
In addition, there is a temporary solution in v4. You will need to roll back to v4 but can do so immediately.
https://github.com/containers/podman/discussions/21539

**Comment by Jasonlhy** [comment]
I installed the podman 5.11 and follow the instruction from https://speakerdeck.com/tnk4on/rosetta-support-podman-v5-dot-1 to enable Rosetta support. The mssql can run on my Mac now
